trackpy/feature.py

```diff
@@ -571,27 +582,51 @@ def batch(frames, diameter, minmass=100, maxsize=None, separation=None,
             # Interpret meta to be a file handle.
             record_meta(meta_info, meta)
 
-    all_features = []
-    for i, image in enumerate(frames):
-        features = locate(image, diameter, minmass, maxsize, separation,
-                          noise_size, smoothing_size, threshold, invert,
-                          percentile, topn, preprocess, max_iterations,
-                          filter_before, filter_after, characterize,
-                          engine)
-        if hasattr(image, 'frame_no') and image.frame_no is not None:
-            frame_no = image.frame_no
-            # If this works, locate created a 'frame' column.
-        else:
-            frame_no = i
-            features['frame'] = i  # just counting iterations
-        logger.info("Frame %d: %d features", frame_no, len(features))
-        if len(features) == 0:
-            continue
+    # Prepare wrapped function for mapping to `frames`
+    curried_locate = partial(
+        locate,
+        diameter=diameter, minmass=minmass, maxsize=maxsize,
+        separation=separation, noise_size=noise_size,
+        smoothing_size=smoothing_size, threshold=threshold, invert=invert,
+        percentile=percentile, topn=topn, preprocess=preprocess,
+        max_iterations=max_iterations, filter_before=filter_before,
+        filter_after=filter_after, characterize=characterize, engine=engine
+    )
+    # Choose threaded or normal map function
+    if threads == 0:
+        pool = None
+        map_func = map
+    else:
+        pool = ThreadPool(processes=threads)
+        map_func = pool.imap
 
-        if output is None:
-            all_features.append(features)
-        else:
-            output.put(features)
+    if report_hook is None:
+        # Use default logger to report process
+        def report_hook(frame_no, detected_features):
+            logger.info("Frame %d: %d features", frame_no, detected_features)
+
+    try:
+        all_features = []
+        for i, features in enumerate(map_func(curried_locate, frames)):
+            image = frames[i]
+            if hasattr(image, 'frame_no') and image.frame_no is not None:
+                frame_no = image.frame_no
+                # If this works, locate created a 'frame' column.
+            else:
+                frame_no = i
+                features['frame'] = i  # just counting iterations
+            report_hook(frame_no, len(features))
```

Comment:
I would go for passing `features` alltogether and doing the `len` inside the `report_hook`. It is in the same process anyway, so we don't pay any price for doing that.

Reply:
Sounds like a good idea.